Doppler.MercadoPagoApi/Controllers/PaymentController.cs

```diff
@@ -1,15 +1,49 @@
+using Doppler.MercadoPagoApi.DopplerSecurity;
+using Doppler.MercadoPagoApi.Models;
+using Doppler.MercadoPagoApi.Services;
+using MercadoPago.Client.Customer;
+using MercadoPago.Error;
 using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Threading.Tasks;
 
 namespace Doppler.MercadoPagoApi.Controllers
 {
     [Authorize]
     [ApiController]
     public class PaymentController : Controller
     {
-        public PaymentController()
+        private readonly IMercadoPagoService _mercadoPagoService;
+
+        public PaymentController(IMercadoPagoService mercadoPagoService)
         {
+            _mercadoPagoService = mercadoPagoService;
+        }
 
+        [Authorize(Policies.OWN_RESOURCE_OR_SUPERUSER)]
+        [HttpPost("/accounts/{accountname}/customer")]
+        public async Task<IActionResult> CreateCustomerAsync([FromRoute] string accountname, CustomerDto customer)
+        {
+            var customerRequest = new CustomerRequest
+            {
+                Email = accountname,
+                FirstName = customer.FirstName,
+                LastName = customer.LastName
+            };
+            try
+            {
+                var savedCustomer = await _mercadoPagoService.CreateCustomerAsync(customerRequest);
+                return Ok(new { CustomerId = savedCustomer.Id });
+            }
+            catch (MercadoPagoApiException exception)
+            {
+                return BadRequest(exception.ApiError);
+            }
```

Comment:
Why is it a Bad Request result? is it a problem with the request? What kinds of errors could throw this exception?

Reply:
This type of exception (MercadoPagoApiException) is thrown when the MercadoPago API response is unsuccessful. In this case, when a customer already exists for example. For unexpected errors the SDK throws MercadoPagoException

Reply:
Ok, BadRequest could be fine. I would prefer a 500 error while we define how to deal with any of the expected errors, but it is opinable.